File: NZSLDict/Classes/ViewControllerPhone.m
//
//  ViewControllerPhone.m
//  NZSL Dict
//
//  Created by Greg Hewgill on 25/04/13.
//
//

#import "ViewControllerPhone.h"

#import "DiagramViewController.h"
#import "SearchViewController.h"
#import "VideoViewController.h"
#import "HistoryViewController.h"
//#import "AboutViewController.h"
#import "NZSL_Dict-Swift.h"

@interface ViewControllerPhone ()

@end

@implementation ViewControllerPhone {
    SearchViewController *searchController;
    DiagramViewController *diagramController;
    VideoViewController *videoController;
    HistoryViewController *historyController;
    UIViewController *aboutController;
}

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view.
    self.viewControllers = @[
        searchController = [[SearchViewController alloc] init],
        diagramController = [[DiagramViewController alloc] init],
        videoController = [[VideoViewController alloc] init],
        historyController = [[HistoryViewController alloc] init],
        aboutController = [[AboutViewController alloc] init],
    ];
    searchController.delegate = self;
    diagramController.delegate = self;
    videoController.delegate = self;
    historyController.delegate = self;
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)toInterfaceOrientation
{
    return toInterfaceOrientation == UIInterfaceOrientationPortrait
        || UIInterfaceOrientationIsLandscape(toInterfaceOrientation);
}

- (void)returnToSearchView
{
    self.selectedViewController = searchController;
}

- (void)didSelectEntry:(DictEntry *)entry
{
    self.selectedViewController = diagramController;
}

@end

File: NZSLDict/Classes/ViewControllerPhone.swift
import UIKit

class ViewControllerPhone: UITabBarController, ViewControllerDelegate, SearchViewControllerDelegate {
    var searchController: SearchViewController!
    var diagramController: DiagramViewController!
    var videoController: VideoViewController!
    var historyController: HistoryViewController!

    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    override func viewDidLoad() {
        super.viewDidLoad()

        searchController = SearchViewController()
        diagramController = DiagramViewController()
        videoController = VideoViewController()
        historyController = HistoryViewController()

        self.viewControllers = [
            searchController,
            diagramController,
            videoController,
            historyController,
        ]

        searchController.delegate = self
        diagramController.delegate = self
        videoController.delegate = self
        historyController.delegate = self
    }

       override var shouldAutorotate : Bool {
        return true
    }

    func didSelectEntry(_ entry: DictEntry) {
        self.selectedViewController = diagramController
    }

    func returnToSearchView() {
        self.selectedViewController = searchController
    }
}
